### PR TITLE
Added C++ language 

### PR DESCRIPTION
**Pull request by anjaliSilawat**
### Description 
I have expanded the language section by adding comprehensive content on C++. It covers various topics such as data types, control structures, functions, classes, object-oriented programming, and memory management. The content is suitable for both beginners and advanced learners.

issue #629 

image

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### CODE REVIEW

**Review by rupali-codes — CHANGES_REQUESTED** [review]
I have addressed a few things:
* please update your code with `linkshub:main`
* the description should be at least 15 words long
* your changes are not reflected on the site, please read & follow our [CONTRIBUTING guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) carefully

**Comment by anjaliSilawat** [comment]
> I have addressed a few things:
> 
> * please update your code with `linkshub:main`
> * the description should be at least 15 words long
> * your changes are not reflected on the site, please read & follow our [CONTRIBUTING guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) carefully

@rupali-codes  I have updated the description also I update my code hope so changes are visible now on the site

**Comment by rupali-codes** [comment]
@anjaliSilawat hey, please resolve the merge conflicts and your vercel deployment is getting bcz the of the following issues, kindly fix them
image

**Comment by CBID2** [comment]
@anjaliSilawat, can you mention the issue number that your PR solves per the [Contributing files](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md#making-pull-requests) please?

**Comment by anjaliSilawat** [comment]
> @anjaliSilawat, can you mention the issue number that your PR solves per the [Contributing files](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md#making-pull-requests) please?

done

**Comment by rupali-codes** [comment]
@anjaliSilawat can you please resolve merge conflicts

**Comment by anjaliSilawat** [comment]
> @anjaliSilawat can you please resolve merge conflicts


Yes, I'll give me a little bit of time. @rupali-codes

**Comment by rupali-codes** [comment]
> > @anjaliSilawat can you please resolve merge conflicts
> 
> Yes, I'll give me a little bit of time. @rupali-codes

Sure, take your time

**Comment by rupali-codes** [comment]
@anjaliSilawat are you ready to work on it now?

File: database/languages/c++.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[
+    {
+        "name": "c++ Tutorial",
+        "description": " Learn the basics of C++ it is a popular programming language.C++ is used to create computer programs, and is one of the most used language in game development.",
```
Comment (CBID2) [inline database/languages/c++.json:4]: [Recommendation] This description mostly describes what C++ is. While helpful, I highly suggest describing **why** people in tech should use this source. 
```suggestion
        "description": " Learn the basics of C++ it is a popular programming language.C++ is used to create computer programs, and is one of the most used languages in game development.",
```

File: database/languages/c++.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        "url": "https://www.geeksforgeeks.org/category/cpp/",
+        "category": "Languages",
+        "subcategory": "C++"
+    },
+    {
+        "name": "C++ Interview Questions",
+        "description": "Top 50 C++ Interview Questions and Answers (2023)",
+        "url": "https://www.geeksforgeeks.org/cpp-interview-questions/",
+        "category": "Languages",
+        "subcategory": "C++"
```
Comment (CBID2) [inline database/languages/c++.json:21]: [Recommendation] Use my feedback in the [previous lines](https://github.com/rupali-codes/LinksHub/pull/957/files#r1234652002) to improve these descriptions. 
```suggestion
        "name": "C++ Recent Articles! ",
        "description": "Recent Articles on C++",
        "url": "https://www.geeksforgeeks.org/category/cpp/",
        "category": "Languages",
        "subcategory": "C++"
    },
    {
        "name": "C++ Interview Questions",
        "description": "Top 50 C++ Interview Questions and Answers (2023)",
        "url": "https://www.geeksforgeeks.org/cpp-interview-questions/",
        "category": "Languages",
        "subcategory": "C++"
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @anjaliSilawat! :) I resolved the merge conflicts from your branch. What's left to do now is revising the descriptions per my advice in my in-line comments(keep this advice in mind as you submit more pull requests to LinksHub).

**Comment by rupali-codes** [comment]
@anjaliSilawat i'm closing this PR bcz you have resolved conflicts improperly, please make fresh PR and before that make sure to update your branch with `linkshub:main`.

**Pull request closed without merging**